erpnext/stock/get_item_details.py

```diff
@@ -101,6 +101,15 @@ def get_item_details(args, doc=None, for_validate=False, overwrite_warehouse=Tru
 
 	frappe.utils.call_hook_method("get_item_details", args, out, doc=doc, for_validate=for_validate)
 
+	if args.get("doctype") in ["Sales Order", "Sales Invoice"]:
```

Comment:
instead of checking for doctype, better to check the line item meta has field `last_billed_rate`

see for reference: https://github.com/ParaLogicTech/erpnext/blob/6cf867a05faeeb472d93e95964ef0ed817125a03/erpnext/stock/get_item_details.py#L354-L357